superbench/benchmarks/micro_benchmarks/directx_gemm_flops_performance/GPUCore.cpp

```diff
@@ -0,0 +1,507 @@
+        for (int i = 0; i < opts->num_warm_up; ++i) {
+            InitializeExecuteComputeOp<uint16_t>(m, n, k);
+        }
+        for (int i = 0; i < loops; ++i) {
+            gpuTimer.init(m_device.Get(), m_commandQueue.Get(), 1, D3D12::QueueType::compute);
+            // Do FLOPs job.
+            double timeInMs = InitializeExecuteComputeOp<uint16_t>(m, n, k);
+            auto flops = (int64_t(m) * n * k + m * n) * 2 * 1e-9 / timeInMs;
+            std::cout << flops << " TFLOPs" << std::endl;
+        }
+    } break;
+    }
+}
+
+/**
+ * @brief Create pipeline including
+ *		  create device object, command list, command queue
+ *		  and synchronization objects.
+ */
+void GPUCore::LoadPipeline() {
+    UINT dxgiFactoryFlags = 0;
+
+#if DEBUG
+    // Enable the debug layer (requires the Graphics Tools "optional feature").
+    // NOTE: Enabling the debug layer after device creation will invalidate the active device.
+    {
+        ComPtr<ID3D12Debug> debugController;
+        if (SUCCEEDED(D3D12GetDebugInterface(IID_PPV_ARGS(&debugController)))) {
+            debugController->EnableDebugLayer();
+
+            // Enable additional debug layers.
+            dxgiFactoryFlags |= DXGI_CREATE_FACTORY_DEBUG;
+        }
+    }
+#endif
+
+    ComPtr<IDXGIFactory4> factory;
+    ThrowIfFailed(CreateDXGIFactory2(dxgiFactoryFlags, IID_PPV_ARGS(&factory)));
+
+    ComPtr<IDXGIAdapter1> hardwareAdapter;
+    GetHardwareAdapter(factory.Get(), &hardwareAdapter);
+
+    // Create GPU device object.
+    ThrowIfFailed(D3D12CreateDevice(hardwareAdapter.Get(), D3D_FEATURE_LEVEL_11_0, IID_PPV_ARGS(&m_device)));
+
+    DML_CREATE_DEVICE_FLAGS dmlCreateDeviceFlags = DML_CREATE_DEVICE_FLAG_NONE;
+
+#if defined(_DEBUG)
+    // If the project is in a debug build, then enable the Direct3D 12 debug layer.
+    // This is optional (starting in DML_FEATURE_LEVEL_5_2) but strongly recommended!
+
+    // If the project is in a debug build, then enable debugging via DirectML debug layers with this flag.
+    dmlCreateDeviceFlags |= DML_CREATE_DEVICE_FLAG_DEBUG;
+#endif
+
+    ThrowIfFailed(DMLCreateDevice(m_device.Get(), dmlCreateDeviceFlags, IID_PPV_ARGS(m_dmlDevice.GetAddressOf())));
+
+    D3D12_COMMAND_QUEUE_DESC queueDesc;
+    // Initialize command queue.
+    ZeroMemory(&queueDesc, sizeof(queueDesc));
+
+    // Describe and create the command queue.
+    queueDesc.Flags = D3D12_COMMAND_QUEUE_FLAG_NONE;
+    queueDesc.Type = D3D12_COMMAND_LIST_TYPE_DIRECT;
+
+    D3D12_COMMAND_QUEUE_DESC cqd3 = {};
+    cqd3.Type = D3D12_COMMAND_LIST_TYPE_DIRECT;
+    m_device->CreateCommandQueue(&cqd3, IID_PPV_ARGS(&m_commandQueue));
+
+    m_device->CreateCommandAllocator(D3D12_COMMAND_LIST_TYPE_DIRECT, IID_PPV_ARGS(&m_commandAllocator));
+
+    m_device->CreateCommandList(0, D3D12_COMMAND_LIST_TYPE_DIRECT, m_commandAllocator.Get(), nullptr,
+                                IID_PPV_ARGS(&m_commandList));
+
+    ThrowIfFailed(m_commandList->Close());
```

Comment:
Why do we need to close and reset it?

Reply:
removed, not necessary, just reset it into a clean recording state to avoid potential issues when entering the main loop